File: problems/2341-maximum-number-of-pairs-in-array/solution.cpp
class Solution {
public:
    vector<int> numberOfPairs(vector<int>& nums) {
        vector<int> count(101, 0);
        for (int num : nums) {
            count[num]++;
        }
        int pairs = 0;
        int leftovers = 0;
        for (int i = 0; i <= 100; i++) {
            pairs += count[i] / 2;
            leftovers += count[i] % 2;
        }
        return {pairs, leftovers};
    }
};

File: problems/2341-maximum-number-of-pairs-in-array/solution.java
import java.util.HashMap;

class Solution {
    public int[] numberOfPairs(int[] nums) {
        HashMap<Integer, Integer> counts = new HashMap<>();
        for (int num : nums) {
            counts.put(num, counts.getOrDefault(num, 0) + 1);
        }
        int pairs = 0;
        int leftovers = 0;
        for (int count : counts.values()) {
            pairs += count / 2;
            leftovers += count % 2;
        }
        return new int[]{pairs, leftovers};
    }
}